### PR TITLE
Add test for controllers

### PR DESCRIPTION
**Pull request by MarvelGa**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -100,6 +100,21 @@
             <artifactId>jjwt</artifactId>
             <version>0.9.1</version>
         </dependency>
+        <dependency>
+            <groupId>com.jayway.jsonpath</groupId>
+            <artifactId>json-path</artifactId>
+            <version>2.4.0</version>
+        </dependency>
+        <dependency>
```
Comment (ahlukhouski2) [inline pom.xml:108]: Is it used?

File: src/main/java/com/exadel/discountwebapp/category/controller/CategoryController.java [hunk 2]
```diff
@@ -29,19 +29,19 @@ public CategoryResponseVO findById(@PathVariable Long id) {
         return categoryService.findById(id);
     }
 
-    @PreAuthorize("hasAnyAuthority('ADMIN')")
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/category/controller/CategoryController.java:32]: @PreAuthorize("isAuthenticated()") ?

File: src/main/java/com/exadel/discountwebapp/exception/WebExceptionHandler.java [hunk 3]
```diff
@@ -61,9 +61,16 @@ public String entityAlreadyExistsException(EntityAlreadyExistsException ex) {
...
-    public String globalException() {
-        return "Something is wrong";
+
+    @ExceptionHandler(AuthenticationException.class)
+    @ResponseStatus(value = HttpStatus.UNAUTHORIZED)
+    public String unauthorizedException(AuthenticationException ex) {
+        return ex.getMessage();
     }
+
+//    @ExceptionHandler(Exception.class)
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/exception/WebExceptionHandler.java]: ?

File: src/main/java/com/exadel/discountwebapp/security/LoginController.java [hunk 4]
```diff
@@ -21,7 +22,7 @@ public TokenResponse signIn(@RequestBody SigninVO signinVO) {
         try {
             userResponse = userService.findByLoginAndPassword(signinVO);
         } catch (EntityNotFoundException e) {
-            throw new EntityNotFoundException("Login or Password is wrong! Try again!");
+            throw new AuthenticationException("Login or Password is wrong! Try again!");
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/security/LoginController.java]: ?

File: src/test/java/com/exadel/discountwebapp/category/controller/CategoryControllerIIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
+@AutoConfigureMockMvc
+@EnableAutoConfiguration(exclude = SecurityAutoConfiguration.class)
+@Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = "classpath:sql/category-init.sql")
+@Sql(executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD, scripts = "classpath:sql/clean-up.sql")
+public class CategoryControllerIIntegrationTest {
```
Comment (ahlukhouski2) [inline src/test/java/com/exadel/discountwebapp/category/controller/CategoryControllerIIntegrationTest.java]: CategoryControllerIIntegrationTest?

File: src/test/java/com/exadel/discountwebapp/category/controller/CategoryControllerIIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
+@AutoConfigureMockMvc
+@EnableAutoConfiguration(exclude = SecurityAutoConfiguration.class)
```
Comment (ahlukhouski2) [inline src/test/java/com/exadel/discountwebapp/category/controller/CategoryControllerIIntegrationTest.java]: Does it work without this exclude?

File: pom.xml (hunk 1 above)
Comment (MarvelGa) [reply pom.xml:108]: I've tested - it is needed

File: src/main/java/com/exadel/discountwebapp/category/controller/CategoryController.java (hunk 2 above)
Comment (MarvelGa) [reply src/main/java/com/exadel/discountwebapp/category/controller/CategoryController.java:32]: can work well without it

**Review by ahlukhouski2 — APPROVED** [review]

**Pull request merged**